dev/devicelab/bin/tasks/plugin_lint_mac.dart

```diff
@@ -59,6 +63,134 @@ Future<void> main() async {
         );
       });
 
+      section('Create Swift plugin');
+
+      const String swiftPluginName = 'test_plugin_swift';
+      await inDirectory(tempDir, () async {
+        await flutter(
+          'create',
+          options: <String>[
+            '--org',
+            'io.flutter.devicelab',
+            '--template=plugin',
+            '--ios-language=swift',
+            swiftPluginName,
+          ],
+        );
+      });
+
+      section('Create Objective-C application');
+
+      const String objcAppName = 'test_app_objc';
+      await inDirectory(tempDir, () async {
+        await flutter(
+          'create',
+          options: <String>[
+            '--org',
+            'io.flutter.devicelab',
+            '--ios-language=objc',
+            objcAppName,
+          ],
+        );
+      });
+
+      section('Build Objective-C application with Swift and Objective-C plugins as libraries');
+
+      final String objcAppPath = path.join(tempDir.path, objcAppName);
+
+      final String swiftPluginPath = path.join(tempDir.path, swiftPluginName);
+      final File objcPubspec = File(path.join(objcAppPath, 'pubspec.yaml'));
+      String podspecContent = objcPubspec.readAsStringSync();
+      podspecContent = podspecContent.replaceFirst(
+        '\ndependencies:\n',
+        '\ndependencies:\n  $objcPluginName:\n    path: $objcPluginPath\n  $swiftPluginName:\n    path: $swiftPluginPath\n',
+      );
+      objcPubspec.writeAsStringSync(podspecContent, flush: true);
+
+      await inDirectory(objcAppPath, () async {
+        await flutter(
+          'build',
+          options: <String>[
+            'ios',
+            '--no-codesign'
+          ],
+        );
+      });
+
+      final File objcPodfile = File(path.join(objcAppPath, 'ios', 'Podfile'));
+      String objcPodfileContent = objcPodfile.readAsStringSync();
+      if (objcPodfileContent.contains('use_frameworks!')) {
+        return TaskResult.failure('Expected default Objective-C Podfile to not contain use_frameworks');
+      }
+
+      section('Build Objective-C application with Swift and Objective-C plugins as frameworks');
+
+      objcPodfileContent = 'use_frameworks!\n' + objcPodfileContent;
+      objcPodfile.writeAsStringSync(objcPodfileContent, flush: true);
+
+      await inDirectory(objcAppPath, () async {
+        await flutter(
+          'build',
+          options: <String>[
+            'ios',
+            '--no-codesign'
+          ],
+        );
+      });
+
```

Comment:
Yes, it will fail if any build commands fail.  Most of these build scenarios fail on master due to CocoaPods or linker issues, which cause the build command to fail.
I think we check the output structure in other tests.